galsim/inclinedexponential.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2012-2016 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""@file inclinedexponential.py
+
+InclinedExponential is a class representing an exponential profile inclined to the LOS.
+"""
+
+import galsim
+from galsim import GSObject
+from . import _galsim
+
+
+class InclinedExponential(GSObject):
+    """A class describing an inclined exponential profile.
+
+    The Inclined Exponential surface brightness profile is characterized by three properties: its
+    inclination angle 'i' (where 0 = face-on and pi/2 = edge-on), its scale radius, and its scale
+    height. The 3D light distribution function is:
+
+    I(R,z) = I_0 / (2h_s) * sech^2 (z/h_s) * exp(-R/R_s)
+
+    where z is the distance along the minor axis, R is the radial distance from the minor axis,
+    R_s is the scale radius of the disk, h_s is the scale height of the disk, and I_0 is the central
+    surface brightness of the face-on disk.
+
+    In this implementation, the profile is inclined along the y-axis. This means that it will likely
+    need to be rotated in most circumstances.
+
+    At present, this profile is not enabled for photon-shooting.
+
+    Initialization
+    --------------
+
+    @param inclination      The inclination angle, i
```

Comment:
Also, state what limitations should be imposed on input.  For example, if I decide to go crazy and  initialize with ```inclination=90.*galsim.degrees``` (perfectly edge on), I get the following output
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "galsim/inclinedexponential.py", line 77, in __init__
    gsparams))
RuntimeError: Solve error: Too many iterations in bracket()
```
Actually the same is true when I try 80 degrees.  This is something that users should be told up-front.

If you prefer not to include such information in the @param statements, then higher up in the docstring you could give an example of how to initialize an instance of this class (this is not very unusual; many classes have something like this) and discuss such limitations there.

It would also be natural for users to wonder about constraints on ```scale_height``` values.